src/ragas/testset/generator.py

```diff
@@ -301,7 +301,7 @@ def generate(
 
         try:
             test_data_rows = exec.results()
-            if not test_data_rows:
+            if all([is_nan(row) for row in test_data_rows]):
                 raise ExceptionInRunner()
```

Comment:
actually I don't know if this change is accurate. Originally if exec.result is `None` it signified an error in executor. Since this is not the case now that could be one change here

Reply:
I agree for your statement, it would raise an error if `exec.result` is None, but currently it returns a list of results:
https://github.com/explodinggradients/ragas/blob/ffebf4f064ba995cbf9652a56183ece7800170d8/src/ragas/executor.py#L109
Python does not handle those values the same way, you can highlight it with:
```python
>>> print("OK" if not None else "Not Ok")
OK
>>> print("OK" if not [None] else "Not Ok")
Not Ok
```

Furthermore, it is not a None as the list element since it is the result of a `ragas.testset.evolutions.Evolution` which returns a `np.nan`:
```python
>>> print("OK" if not np.nan else "Not Ok")
Not Ok
>>> print("OK" if not [np.nan] else "Not Ok")
Not Ok
```

Reply:
In any case, thanks again for your amazing work that helps me and a lot of the research community! I'm just trying to make my contribution to this wonderful project so I don't fix it only on my side but also on everybody side!

Reply:
hey @Gwenn-LR thanks for the kind words 🙂